shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -71,17 +79,42 @@ public void run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI srcPath = new AlluxioURI(args[0]);
     AlluxioURI dstPath = new AlluxioURI(args[1]);
-    List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
-    if (srcPaths.size() == 0) {
-      throw new FileDoesNotExistException(
-          ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath.getPath()));
-    }
+    if (dstPath.getScheme() == null && "file".equals(srcPath.getScheme())) {
+      List<File> srcFiles = AlluxioShellUtils.getFiles(srcPath.getPath());
+      if (srcFiles.size() == 0) {
+        throw new IOException("Local path " + srcPath + " does not exist.");
+      }
+      if (srcPath.getPath().contains(AlluxioURI.WILDCARD)) {
+        copyFromLocalWildcard(srcFiles, dstPath);
+      } else {
+        copyFromLocal(new File(srcPath.getPath()), dstPath);
+      }
+    } else if (srcPath.getScheme() == null && "file".equals(dstPath.getScheme())) {
+      File dstFile = new File(args[1]);
+      List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
+      if (srcPaths.size() == 0) {
+        throw new IOException(srcPath.getPath() + " does not exist.");
+      }
 
-    if (srcPath.containsWildcard()) {
-      copyWildcard(srcPaths, dstPath, cl.hasOption("R"));
+      if (srcPath.containsWildcard()) {
+        copyWildcardToLocal(srcPaths, dstFile);
+      } else {
+        copyToLocal(srcPath, dstFile);
+      }
     } else {
-      copy(srcPath, dstPath, cl.hasOption("R"));
+      List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
```

Comment:
And what happens when src or dst has schemes other than `file` like `s3` or `hdfs`?